items.c

```diff
@@ -109,13 +110,18 @@ static uint64_t lru_total_bumps_dropped(void);
 /* Get the next CAS id for a new item. */
 /* TODO: refactor some atomics for this. */
 uint64_t get_cas_id(void) {
-    static uint64_t cas_id = 0;
     pthread_mutex_lock(&cas_id_lock);
     uint64_t next_id = ++cas_id;
     pthread_mutex_unlock(&cas_id_lock);
     return next_id;
 }
 
+void set_cas_id(uint64_t new_cas) {
+    pthread_mutex_lock(&cas_id_lock);
```

Comment:
Exactly (think I even had a comment about that somewhere), but I want that to be a deliberate/separate patch. Since it doesn't block this particular PR and can be fiddly with RISC portability I haven't done it (specifically there's a PA-RISC bug probably related to atomics that I haven't found yet).